// @clerk/shared/dist/cjs/hooks/useOrganizationList.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var useOrganizationList_exports = {};
__export(useOrganizationList_exports, {
  useOrganizationList: () => useOrganizationList
});
module.exports = __toCommonJS(useOrganizationList_exports);
var import_contexts = require("./contexts");
const useOrganizationList = () => {
  const clerk = (0, import_contexts.useClerkInstanceContext)();
  const user = (0, import_contexts.useUserContext)();
  if (!clerk.loaded || !user) {
    return { isLoaded: false, organizationList: void 0, createOrganization: void 0, setActive: void 0 };
  }
  return {
    isLoaded: true,
    organizationList: createOrganizationList(user.organizationMemberships),
    setActive: clerk.setActive,
    createOrganization: clerk.createOrganization
  };
};
function createOrganizationList(organizationMemberships) {
  return organizationMemberships.map((organizationMembership) => ({
    membership: organizationMembership,
    organization: organizationMembership.organization
  }));
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  useOrganizationList
});
